#### Python Operator Assignment
#### This notebook contains solutions to Python Operator Assignment questions.
#### ==============================================================

Q.1) What is the output of the following addition (+) operator

a = [10, 20]
b = a
b += [30, 40]

print(a)
print(b)


In [1]:
a = [10, 20]
b = a
b += [30, 40]

In [2]:
print(a)

[10, 20, 30, 40]


In [3]:
print(b)

[10, 20, 30, 40]


##### Answer: a and b both contain [10, 20, 30, 40].
##### Conclusion:Since b = a, both variables refer to the same list object. The += operator modifies the list in place, so the changes made through b are also reflected in a. Therefore, both a and b become: [10, 20, 30, 40].
##### ===========================================================================

Q.2) What is the output of print(2%6)

a)	  ValueError
b)	  0.33
c)	  2


In [4]:
print(2%6)

2


##### Answer: (c) 2
#### =====================================================================

Q.3) What is the output of the following code:
print(bool(0), bool(3.14159), bool(-3), bool(1.0+1j))

a)	  True True False True
b)	  False True True True
c)	  True True False True
d)	  False True False True


In [5]:
print(bool(0), bool(3.14159), bool(-3), bool(1.0+1j))


False True True True


##### Answer: (b) False True True True
#### =====================================================================

Q.4) What is the output of the expression  print(-18 // 4) ?

In [6]:
print(-18 // 4) 

-5


##### Explaination: -18 / 4 = -4.5, floor value of -4.5 is -5
##### Conclusion: The // operator returns the floor value of the division result. Since -18 ÷ 4 = -4.5, Python rounds down to -5. Therefore, the output is -5.
#### =====================================================================

Q.5) What is the output of print(2 ** 3 ** 2) ?

In [7]:
print(2 ** 3 ** 2) 

512


##### Explaination: 3 ** 2 = 9,2 ** 9 = 512
##### Conclusion: Python evaluates exponentiation (**) from right to left. Therefore, 2 ** 3 ** 2 is calculated as 2 ** (3 ** 2), which equals 512.
#### ===================================================================

Q.6) 4 is 100 in binary and 11 is 1011. What is the output of the following bitwise operators?

a = 4
b = 11
print(a | b)
print(a >> 2)


In [9]:
a = 4
b = 11
print(a | b)
print(a >> 2)

15
1


##### Answer: a | b performs a bitwise OR operation and returns 15.
##### a >> 2 shifts the bits of 4 two places to the right and returns 1.
#### =====================================================================

Q.7) What is the value of the following Python Expression

print(36 / 4)


In [10]:
print(36 / 4)

9.0


##### Answer: 36 / 4 = 9.0
#### =============================================================

Q.8) What is the output of the following assignment operator

y = 10
y+= 2
x = y
print(x)


In [16]:
y = 10
y+= 2
x = y 
print(x)


12


##### Answer: 12 is the value of x 
#### =====================================================================

Q.9) What is the output of print(2 * 3 ** 3 * 4)

In [17]:
print(2 * 3 ** 3 * 4)

216


##### Answer: Python evaluates the exponent first (3³ = 27), then performs the multiplications. Therefore, the result of 2 * 3 ** 3 * 4 is 216.
#### =====================================================================

Q.10) Which of the following operators has the highest precedence

#### Order of precedence (highest to lowest):
##### 1. ** (Exponentiation)
##### 2. *, /, //, %
##### 3. +, -

##### Conclusion:-Among the common arithmetic operators (+, -, *, /, //, %, **), the exponentiation operator (**) has the highest precedence.
#### =================================================================

Q.11) What is the output of print(10 - 4 * 2) ?

In [20]:
print(10 - 4 * 2) 

2


##### Answer: first when we do 4*2=8, then minus 10 with 8 the answer is 2,which is 10-8=2
#### =========================================================================

Q.12) What is the output of the following code:

x = 6
y = 2

print(x ** y)
print(x // y)


In [21]:
x = 6
y = 2


In [22]:
print(x ** y)


36


In [23]:
print(x // y)


3


##### Answer of x ** y = 6 ** 3= is 36, and x // y = 6//3 is 3 
#### ===============================================================

## <span style="color:purple">Conclusion</span>
In this assignment, I practiced: Operators
- Arithematic Operators
- Bitwise Operators
- operator precedence
- Boolean Values
- Assignment Operators